### PR TITLE
Add explicit dependencies on several tools we need

### LINKED ISSUE #1335: nix 1.12 assumes that lsof is installed
nix-lsof

### PR DESCRIPTION
**Pull request by copumpkin**
I left unzip and git implicit, but made sure they had loud comments next to them.

Should fix #1335, #1328, and #1234 

Also, regarding the autoconf,

no idea

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -11,6 +11,7 @@ with import <nixpkgs> {};
       pkgconfig sqlite libsodium boehmgc
       docbook5 docbook5_xsl
       autoconf-archive
+      lsof # Used by the garbage collector to determine what's in use
```
Comment (copumpkin) [inline shell.nix]: The duplication between here and release.nix made me uncomfortable but they do seem pretty duplicatey already.

**Comment by domenkozar** [comment]
What if Nix is compiled on different machine than it's used?

**Comment by copumpkin** [comment]
@domenkozar yes, it would include store paths for those tools whenever built by Nix. And if someone builds it on a non-Nix system we'll just embed non-store paths and nothing is lost.

@edolstra any thoughts? I saw that since I submitted this PR, you already did `lsof` a little differently, and are not depending on `lsof` in `release.nix`. That will probably work in practice on macOS, but it'd be nice to depend explicitly on the package.

**Comment by fricklerhandwerk** [comment]
@copumpkin I think it's unfortunate the bot was closing pull requests. Are you still interested in working on this? The addressed issues are still hitting Nix beginners.

**Pull request closed without merging**